src/algebra/group/hom.lean

```diff
@@ -134,6 +134,16 @@ lemma exists_inv_of_comp_exists_inv {S : set M} {f : M →* N}
   ∃ x : P, g.comp f s * x = 1 :=
 let ⟨c, hc⟩ := hf s H in ⟨g c, show g _ * _ = _, by rw [←g.map_mul, hc, g.map_one]⟩
 
+@[to_additive]
+lemma cancel_right {g₁ g₂ : N →* P} {f : M →* N} (hf : function.surjective f) :
+  g₁.comp f = g₂.comp f ↔ g₁ = g₂ :=
+⟨λ h, monoid_hom.ext $ (forall_iff_forall_surj hf).1 (ext_iff.1 h), λ h, h ▸ rfl⟩
+
+@[to_additive]
+lemma cancel_left {g : N →* P} {f₁ f₂ : M →* N} (hg : function.injective g) :
+  g.comp f₁ = g.comp f₂ ↔ f₁ = f₂ :=
+⟨λ h, monoid_hom.ext $ λ x, hg $ by rw [← comp_apply, h, comp_apply], λ h, h ▸ rfl⟩
```

Comment:
Maybe also do this for ring_homs, while you're at it?

Reply:
Done.